src/linear_algebra/invariant_submodule.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import linear_algebra.basic
+import linear_algebra.projection
+
+/-!
+# Invariant submodules
+
+In this file, we define and prove some basic results on invariant submodules.
+-/
+
+namespace submodule
+
+variables {E R : Type*} [ring R] [add_comm_group E] [module R E]
+
+/-- `U` is `T` invariant (ver 1): `U ≤ U.comap` -/
+def invariant_under (U : submodule R E) (T : E →ₗ[R] E) : Prop := U ≤ U.comap T
+
+/-- `U` is `T` invariant if and only if `U.map T ≤ U` -/
+@[simp] lemma invariant_under_iff_map (U : submodule R E) (T : E →ₗ[R] E) :
```

Comment:
I don't think this is a good `simp` lemma; there's not much point having a definition if `simp` immediately replaces it with something else.